scripts/packages/publish-to-local-registry.ts

```diff
@@ -0,0 +1,72 @@
+import execa from "execa";
+import sleep from "sleep-promise";
+import path from "path";
+import { logStepEnd, logStepStart } from "../shared/logging";
+
+// These variables are hardcoded on purpose. We don’t want to publish to a real registry by mistake.
+const npmRegistry = "http://localhost:4873";
+const npmUserAndPassword = "verdaccio";
+const npmEmail = "verdaccio@example.com";
+
+const packageNames = [
+  "block-template",
+  "blockprotocol",
+  "create-block-app",
+  "mock-block-dock",
+];
+
+const defaultExecaOptions = {
+  env: {
+    NODE_ENV: "development",
+    NPM_CONFIG_REGISTRY: npmRegistry,
+    PATH: process.env.PATH,
+  },
+  extendEnv: false,
+  stdio: "inherit",
+} as const;
+
+const script = async () => {
+  logStepStart("Login into local registry");
+
```

Comment:
Could potentially bail here if `defaultExecaOptions.env.NPM_CONFIG_REGISTRY` doesn't contain `"localhost"`, and warn people that this script will _unpublish_ things too, just in case 😅  - I'm thinking of someone coming along and thinking 'oh a script to publish the packages, I'll change the config to publish for real'

Reply:
We use `NPM_CONFIG_REGISTRY: process.env.NPM_CONFIG_REGISTRY` in [scripts/packages/smoke-test-create-block-app.ts](https://github.com/blockprotocol/blockprotocol/pull/178/files#diff-633ed29d74de8a8abed1965dc8ffb2151fdece412f0678592e1b2534098c7856). In the initial version of this script we have a hardcoded value: `NPM_CONFIG_REGISTRY: npmRegistry`.

Does this address your concern?

Reply:
Not really - my concern is that the value in `defaultExecaOptions` are changed, and the script run. 

The function itself has nothing to prevent it being run using different options, which might access the real registry, and unpublish a package.

Reply:
Oh IC. Right, I can change that. So we’ll have hardcoded `npmRegistry` plus early exit if the value is not `localhost:N`. Makes sense!

Reply:
It's protecting against a very unlikely event, but since there's a near-zero cost to doing so I'd feel better for it - thank you!

Reply:
https://github.com/blockprotocol/blockprotocol/pull/178/commits/e43481817c1b5daff119526aba51c49c0c4934cc